filebeat/registrar/registrar.go

```diff
@@ -62,69 +66,26 @@ var (
 	registrySuccess = monitoring.NewInt(nil, "registrar.writes.success")
 )
 
+const fileStatePrefix = "filebeat::logs::"
+
 // New creates a new Registrar instance, updating the registry file on
 // `file.State` updates. New fails if the file can not be opened or created.
-func New(cfg config.Registry, out successLogger) (*Registrar, error) {
-	home := paths.Resolve(paths.Data, cfg.Path)
-	migrateFile := cfg.MigrateFile
-	if migrateFile != "" {
-		migrateFile = paths.Resolve(paths.Data, migrateFile)
-	}
-
-	err := ensureCurrent(home, migrateFile, cfg.Permissions)
+func New(stateStore StateStore, out successLogger, flushTimeout time.Duration) (*Registrar, error) {
```

Comment:
Note (no need to fix it): Probably no one uses the state store besides us. However, in theory, it would be more elegant to not to change the exported `New` function. But rather add a deprecation warning. The new function could be called maybe `NewFromStateStore` or whatever.

Reply:
TBH the registrar package will be removed. We first need to change the log input to the new API though. So I wasn't too concerned about breaking the API (plus registrar is an implementation detail).

Reply:
OK.